spec/unit/provider/service/launchd_spec.rb

```diff
@@ -242,6 +246,28 @@
     end
   end
 
+  describe "when a service is unavailable" do
+    let(:map) { {"some.random.job" => "/path/to/job.plist"} }
+
+    it "should fail when enabling the service and enable should be :false" do
+      allow(provider).to receive(:make_label_to_path_map).and_return(map)
+      expect { provider.jobsearch("NOSUCH") }.to raise_error(Puppet::Error)
+      expect(subject.enabled?).to eq(:false)
+    end
+
+    it "should fail when starting the service" do
+      allow(provider).to receive(:make_label_to_path_map).and_return(map)
+      expect { provider.jobsearch("NOSUCH") }.to raise_error(Puppet::Error)
```

Comment:
you can leave just the `expect { subject.start }.to raise_error(Puppet::Error)` since this part is tested in test from :252